meerkat/model/model.py

```diff
@@ -0,0 +1,127 @@
+from typing import Dict, List
+
+import torch
+
+from meerkat import DataPanel
+from meerkat.columns.prediction_column import ClassificationOutputColumn
+from meerkat.model.metrics import compute_metric
+
+
+# TODO(Priya): Move some general functions here
+class Model:
+    def __init__(
+        self,
+        # identifier: str,
+        model,
+        # evaluation_fn=None,
+        is_classifier: bool = None,
+        task: str = None,
+        device: str = None,
+    ):
+
+        # self.identifier = identifier
+
+        self.model = model
+
+        # if evaluation_fn is not None:
+        #    self.evaluate = evaluation_fn
+
+        if task is None:
+            if is_classifier is None or not is_classifier:
+                raise ValueError("Task is required for non-classification models.")
+        else:
+            is_classifier = False
+
+        self.is_classifier = is_classifier
+        self.task = task
+
+        if not device:
+            self.device = "cpu"
+            if torch.cuda.is_available():
+                self.device = "cuda:0"
+
+    def to(self, device: str):
+        self.device = device
+        return self.model.to(device)
+
+    """
```

Comment:
`__call__ ` in `nn.Module` typically just calls forward. If we subclass `nn.Module` we can just remove this commented out block